### PR TITLE
Make kubetest use boskos to get projects if PROJECT is not set from env

### PR DESCRIPTION
**Pull request by krzyzacy**
ref #2202

### CODE REVIEW

**Comment by fejta** [comment]
/cc @mikedanese 

He expressed excitement about this today

File: kubetest/main.go [hunk 1]
```diff
@@ -405,7 +416,14 @@ func prepareGcp(kubernetesProvider string) error {
...
 	p := os.Getenv("PROJECT")
 	if p == "" {
-		return fmt.Errorf("KUBERNETES_PROVIDER=%s requires setting PROJECT", kubernetesProvider)
+		p, err := boskos.Acquire("project", "free", "busy")
+		if err != nil {
+			return fmt.Errorf("KUBERNETES_PROVIDER=%s boskos fail to acquire PROJECT:%s", kubernetesProvider, err)
+		}
+
+		go func(c *client.Client, proj string) {
+			c.UpdateOne(p, "busy")
```
Comment (fejta) [inline kubetest/main.go]: Shouldn't this be inside a for loop or something?

Comment (krzyzacy) [inline kubetest/main.go]: hey well yeah

File: kubetest/main.go [hunk 2]
```diff
@@ -405,7 +416,18 @@ func prepareGcp(kubernetesProvider string) error {
...
 	if p == "" {
-		return fmt.Errorf("KUBERNETES_PROVIDER=%s requires setting PROJECT", kubernetesProvider)
+		p, err := boskos.Acquire("project", "free", "busy")
+		if err != nil {
+			return fmt.Errorf("KUBERNETES_PROVIDER=%s boskos fail to acquire PROJECT:%s", kubernetesProvider, err)
+		}
+
+		go func(c *client.Client, proj string) {
+			for range time.Tick(time.Minute * 5) {
+				if err := c.UpdateOne(p, "busy"); err != nil {
```
Comment (fejta) [inline kubetest/main.go]: Did this keep track of it's owned? Not sure what will happen after the release all call

**Comment by fejta** [comment]
/lgtm

See race question though

File: kubetest/main.go (hunk 2 above)
Comment (krzyzacy) [inline kubetest/main.go]: release is almost the last call from main... so I assume it's fine?

Comment (fejta) [inline kubetest/main.go]: Not at the volume of testing we do... We'll hit the race probably daily at least.

Comment (krzyzacy) [inline kubetest/main.go]: UpdateOne will check if the updated resource is still hold by the client - https://github.com/kubernetes/test-infra/blob/master/boskos/client/client.go#L115-L122, so if it's called after ReleaseAll() it should return a no resource error, which is probably still fine.

Comment (fejta) [inline kubetest/main.go]: Ok

Comment (fejta) [inline kubetest/main.go]: Is the client thread safe then?

Comment (krzyzacy) [inline kubetest/main.go]: hey no :-( fixing

Comment (krzyzacy) [inline kubetest/main.go]: fixed in #2667 , making the image for the canary run

**Comment by krzyzacy** [comment]
let fun time begin, should not affect any existing jobs.

**Pull request merged**